src/main/java/org/hydev/mcpm/utils/Sugar.java

```diff
@@ -54,4 +56,29 @@ public static <T> T[] subTo(T[] arr, int end)
     {
         return sub(arr, 0, end);
     }
+
+    /**
+     * Create a hashmap of values. Note: This function uses an unchecked cast, which assumes that the caller is giving
+     * values of the correct type to the function.
+     *
+     * @param values Map content (Precondition: len(values) % 2 == 0, and each key K is followed by its value V)
+     * @param <K> Type of the key
+     * @param <V> Type of the value
+     * @return Hashmap of values
+     */
+    public static <K, V> Map<K, V> uncheckedMapOf(Object... values)
+    {
+        var map = new HashMap<>();
+        for (int i = 0; i < values.length; i += 2)
+        {
+            var k = values[i];
+            var v = values[i + 1];
+
+            assert k != null;
+            if (v != null) map.put(k, v);
+        }
+
+        //noinspection unchecked
+        return (Map<K, V>) map;
```

Comment:
Do we have to suppress any warnings here?

Reply:
it's an unchecked cast. Due to type erasure, we cannot check if the keys or values are objects of type K or V, we can only put it in the precondition of the function and assume it's true. But it's still way better than Java's `Map.of` that only supports up to 10 pairs because their key and values are hardcoded